File: VocaDbModel/DataContracts/Comments/EntryWithCommentsContract.cs
#nullable disable

using VocaDb.Model.DataContracts.Api;

namespace VocaDb.Model.DataContracts.Comments
{
	public class EntryWithCommentsContract
	{
		public EntryWithCommentsContract() { }

		public EntryWithCommentsContract(EntryForApiContract entry, CommentForApiContract[] comments)
		{
			Comments = comments;
			Entry = entry;
		}

		public CommentForApiContract[] Comments { get; init; }

		public EntryForApiContract Entry { get; init; }
	}
}

File: VocaDbWeb/Scripts/DataContracts/EntryWithCommentsContract.ts
import { CommentContract } from '@/DataContracts/CommentContract';
import { EntryContract } from '@/DataContracts/EntryContract';

// Corresponds to the EntryWithCommentsContract class in C#.
export interface EntryWithCommentsContract {
	comments: CommentContract[];
	entry: EntryContract;
}
